### PR TITLE
L2-715: Ledger with candid

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Ledger Canister uses Candid interface by default. Proto for hardware wallet.

# Changes

* Add Ledger Candid interface and related files and types.
* Extend LedgerCanisterOptions when creating the canister.
* Create the ledger service from candid interface when creating a LedgerCanister.
* Using the ledger service instead of proto by default in "accountBalance" and "transfer" methods.

# Tests

* Add tests for "accountBalance" method both for proto and candid.
* Add test for "transfer" with candid service.

### CODE REVIEW

**Comment by lmuntaner** [comment]
I tested it with a branch in nns-dapp and all works. Also staking a neuron, creating a canister and topping up a canister.

@mstrasinskis @peterpeterparker please review

File: candid/ledger.idl.d.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  | { BadFee: { expected_fee: Tokens } }
+  | { TxDuplicate: { duplicate_of: BlockIndex } }
+  | { TxCreatedInFuture: null }
+  | { InsufficientFunds: { balance: Tokens } };
+export interface TransferFee {
+  transfer_fee: Tokens;
+}
+export type TransferFeeArg = {};
+export type TransferResult = { Ok: BlockIndex } | { Err: TransferError };
+export interface _SERVICE {
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply candid/ledger.idl.d.ts]: This name is done when creating the types from the cancid file. It's also in `governanceTypes.d.ts`.

That said, I refactored to types to follow the same pattern as governance. Where this `_SERVICE` is renamed when exported.

[the remark this answers is not in the document]
Comment (mstrasinskis) [reply candid/ledger.idl.d.ts]: Thanks for the explanation.

**Review by mstrasinskis — APPROVED** [review]
LGTM, thanks.

**Pull request merged**